File: trunk/openrest4j-api/src/com/googlecode/openrest/Category.java
package com.googlecode.openrest;

import java.io.Serializable;

import org.codehaus.jackson.annotate.JsonIgnoreProperties;
import org.codehaus.jackson.map.annotate.JsonSerialize;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Category implements Serializable {
    public Category(String id, String restaurantId, String title, String description,
            String parentCategoryId, String tagId, Double priority) {
        this.id = id;
        this.restaurantId = restaurantId;
        this.title = title;
        this.description = description;
        this.parentCategoryId = parentCategoryId;
        this.tagId = tagId;
        this.priority = priority;
    }

    /** Default constructor for JSON deserialization. */
    public Category() {}

    /** The category's unique id. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String id;

    /** The restaurant's id. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String restaurantId;

    /** The category's title. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String title;

    /** The category's description. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String description;

    /** The parent category's id. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String parentCategoryId;

    /** The item-ids in this category. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
    public String tagId;

    /** Order priority. Higher means first in the list. */
    @JsonSerialize(include = JsonSerialize.Inclusion.NON_DEFAULT)
    public Double priority = 0.0;
    
    private static final long serialVersionUID = 1L;
}

File: trunk/openrest4net/openrest4net-api/com/googlecode/openrest/Category.cs
using System;
using System.Collections.Generic;

namespace com.googlecode.openrest
{
    class Category
    {
        public Category(string id, string restaurantId, string title, string description,
            string parentCategoryId, IList<string> itemIds, double priority)
        {
            this.id = id;
            this.restaurantId = restaurantId;
            this.title = title;
            this.description = description;
            this.parentCategoryId = parentCategoryId;
            this.itemIds = itemIds;
            this.priority = priority;
        }

        /** Empty constructor required for initialization from JSON-encoded string. */
        internal Category() { }

        /** The category's unique id. */
        public string id;

        /** The restaurant's id. */
        public string restaurantId;

        /** The category's title. */
        public string title;

        /** The category's description. */
        public string description;

        /** The parent category's id. */
        public string parentCategoryId;

        /** The item-ids in this category. */
        public IList<string> itemIds = new List<string>();

        /** Order priority. Higher means first in the list. */
        public double? priority = 0.0;
    }
}
